### PR TITLE
Auto create custom search attributes fields in reindex script

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Auto create custom search attributes fields in reindex script.

**Why?**
Better automation for reindex process.

**How did you test it?**
Run manually.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: schema/elasticsearch/visibility/versioned/v1/reindex.sh [hunk 1]
```diff
@@ -13,7 +13,27 @@ ES_VIS_INDEX_V1="${ES_VIS_INDEX_V1:-temporal_visibility_v1_dev}"
...
 AUTO_CONFIRM="${AUTO_CONFIRM:-}"
 
+ES_ENDPOINT="${ES_SCHEME}://${ES_SERVER}:${ES_PORT}"
 DIR_NAME="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
+echo "=== Step 1. Add custom search attributes to the new index. ==="
+# _doc and ?include_type_name=true for compatibility with ES6.
+CUSTOM_SEARCH_ATTRIBUTES_MAPPING=$(curl --silent --user "${ES_USER}":"${ES_PWD}" "${ES_ENDPOINT}/${ES_VIS_INDEX_V0}/_doc/_mapping?include_type_name=true" | jq -c '.. | .Attr? | select(.!=null) | .properties | with_entries(select(.key == ($customSA[]))) | {properties:.}' --argjson customSA "${CUSTOM_SEARCH_ATTRIBUTES}")
+jq -n "${CUSTOM_SEARCH_ATTRIBUTES_MAPPING}"
+if [ -z "${AUTO_CONFIRM}" ]; then
```
Comment (underrun) [inline schema/elasticsearch/visibility/versioned/v1/reindex.sh:23]: nit: prefer `[[` over `[` generally as quoting is easier in many cases

**Review by underrun — APPROVED** [review]

File: schema/elasticsearch/visibility/versioned/v1/reindex.sh (hunk 1 above)
Comment (alexshtin) [reply schema/elasticsearch/visibility/versioned/v1/reindex.sh:23]: Good to know. Leaving it as is for now.

File: schema/elasticsearch/visibility/versioned/v1/reindex.sh [hunk 2]
```diff
@@ -13,7 +13,27 @@ ES_VIS_INDEX_V1="${ES_VIS_INDEX_V1:-temporal_visibility_v1_dev}"
...
+    REPLY="y"
+fi
+curl --silent --user "${ES_USER}":"${ES_PWD}" -X PUT "${ES_ENDPOINT}/${ES_VIS_INDEX_V1}/_doc/_mapping?include_type_name=true" -H "Content-Type: application/json" --data-binary "${CUSTOM_SEARCH_ATTRIBUTES_MAPPING}" --write-out "\n" | jq
+
+until curl --silent --user "${ES_USER}":"${ES_PWD}" "${ES_ENDPOINT}/_cluster/health/${ES_VIS_INDEX_V1}" | jq  --exit-status '.status=="green" | .'; do
+    echo "Waiting for Elasticsearch index ${ES_VIS_INDEX_V1} become green."
+    sleep 1
+done
+
+echo "=== Step 2. Reindex old index to the new index. ==="
```
Comment (yiminc) [inline schema/elasticsearch/visibility/versioned/v1/reindex.sh:36]: do you need to check $(REPLY) before proceeding?

**Review by yiminc — APPROVED** [review]

File: schema/elasticsearch/visibility/versioned/v1/reindex.sh (hunk 2 above)
Comment (alexshtin) [reply schema/elasticsearch/visibility/versioned/v1/reindex.sh:36]: :+1:

**Pull request merged**